### DISCUSSION TITLE
Which dataset is photon using?

### DISCUSSION THREAD

**Discussion by kokoskiwi** [komoot/photon#662] [Q&A] [answered]
Hi are you using https://openaddresses.io/ ? Or are you using another dataset? I didn't found another dataset for addresses; on openstreetmap i can only download the entire planet :D

**Comment by mtmail** [comment, marked as answer]
OpenStreetMap data. The first sentence of the https://github.com/komoot/photon#readme explains it.

**Reply by kokoskiwi** [reply]
> OpenStreetMap data. The first sentence of the https://github.com/komoot/photon#readme explains it.

thanks bro but where exactly are you downloading the addresses from openstreetmap?

> on openstreetmap i can only download the entire planet :D

sorry for asking such a dumb question (._.) i just can't find it. I need only the addresses and want to import them into typesense for building my own search

**Reply by mtmail** [reply]
See the "If you need search data in other languages or restricted to a country you will need to" section in the readme file.

It requires setting up another geocoder (Nominatim) and importing extracts of the OpenStreetMap planet data into Nominatim first. Lots of people use the pre-generated [downloads.geofabrik.de](http://download.geofabrik.de/) extracts but there's other tools for creating custom extracts, e.g. if you just need a single city (https://wiki.openstreetmap.org/wiki/Planet.osm#Worldwide_extract_sources).

Neither Nominatim or Photon have an easy "just give me a list of addresses" mode. That said Nominatim, once installed, has a 'nominatim export' PHP script which I think is undocumented.

https://osmnames.org/download/ shows a one-liner how to cut down their planet data into one country, maybe that's closer to what you need for your project.
